sql/teradata/TeradataSQLDataTypesParser.g4

```diff
@@ -0,0 +1,169 @@
+parser grammar TeradataSQLDataTypesParser;
+
+import TeradataSQLLiteralsParser
+     , TeradataSQLIdentifiersParser
+     , TeradataSQLExpressionsParser
+     ;
+
+options {
+    tokenVocab=TeradataSQLLexer;
+}
+
+data_type
+    : numeric_data_type
+    | char_data_type
+    | binary_data_type
+    | datetime_type
+    | period_type
+    | udt_type
+    ;
+
+variable_data_type
+    : numeric_data_type
+    | char_data_type
+    | precisionless_char_data_type
+    | binary_data_type
+    | datetime_type
+    | period_type
+    | udt_type
+    ;
+
+external_function_data_type
+    : numeric_data_type
+    | char_data_type
+    | precisionless_char_data_type
+    | lob_as_locator_data_type
+    | binary_data_type
+    | datetime_type
+    | period_type
+    | udt_type
+    | TD_ANYTYPE
+    | VARIANT_TYPE
+    ;
+
+numeric_data_type
+    : BYTEINT
+    | SMALLINT
+    | (INTEGER|INT)
```

Comment:
Parentheses are used in places that don't need them. Why? E.g., `(INTEGER | INT)` instead of `INTEGER | INT`.

Reply:
Fixed.